src/DynExpManager/Instruments/NP_Conex_CC.cpp

```diff
@@ -0,0 +1,370 @@
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "RS"; // Go to NOT REFERENCED state.
+			std::this_thread::sleep_for(std::chrono::milliseconds(500)); // This takes 500 ms.
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "PW1"; // Go to CONFIGURATION state.
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "HT1"; // Set current position to home position.
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "PW0"; // Go to NOT REFERENCED state.
+			std::this_thread::sleep_for(std::chrono::seconds(3)); // This takes 3 s.
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "OR"; // Go to READY state.
+		}
+
+		InitFuncImpl(dispatch_tag<InitTask>(), Instance);
+	}
+
+	void NP_Conex_CC_Tasks::ExitTask::ExitFuncImpl(dispatch_tag<PositionerStageTasks::ExitTask>, DynExp::InstrumentInstance& Instance)
+	{
+		ExitFuncImpl(dispatch_tag<ExitTask>(), Instance);
+
+		auto InstrData = DynExp::dynamic_InstrumentData_cast<NP_Conex_CC>(Instance.InstrumentDataGetter());
+
+		try
+		{
+			// Abort motion.
+			*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "ST";
+		}
+		catch (...)
+		{
+			// Swallow any exception which might arise from HardwareAdapter->operator<<() since a failure
+			// of this function is not considered a severe error.
+		}
+
+		Instance.UnlockObject(InstrData->HardwareAdapter);
+	}
+
+	void NP_Conex_CC_Tasks::UpdateTask::UpdateFuncImpl(dispatch_tag<PositionerStageTasks::UpdateTask>, DynExp::InstrumentInstance& Instance)
+	{
+		try
+		{
+			auto InstrData = DynExp::dynamic_InstrumentData_cast<NP_Conex_CC>(Instance.InstrumentDataGetter());
+			auto Owner = DynExp::dynamic_Object_cast<NP_Conex_CC>(&Instance.GetOwner());
+			bool UpdateError = false;
+
+			try
+			{
+				// Tell status:
+				*InstrData->HardwareAdapter << Util::ToStr(InstrData->GetChannel()) + "TS";
+				std::stringstream StatusStream(NP_Conex_CC::AnswerToNumberString(InstrData->HardwareAdapter->WaitForLine(1, std::chrono::milliseconds(25)), "TS"));
+				StatusStream.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+
+				uint16_t ErrorMap; // Variable to hold the error map (16 bits)
+				char ErrorMapBuffer[5]{ 0 };  // 4 characters + 1 for null terminator
+				StatusStream.read(ErrorMapBuffer, 4);  // Read the first 4 characters
+				std::istringstream ErrorMapStream(ErrorMapBuffer);
+
+				ErrorMapStream >> std::hex >> ErrorMap;
+				InstrData->ErrorCode = static_cast<NP_Conex_CC_StageData::ErrorCodeType>(ErrorMap); // Only 0 is no error
+
+				uint16_t State;     // Variable to hold the state (8 bits)
+				char StateBuffer[3]{ 0 };  // 2 characters + 1 for null terminator
+				StatusStream.read(StateBuffer, 2);  // Read the next 2 characters
+				std::istringstream StateStream(StateBuffer);
+
+				StateStream >> std::hex >> State;
```

Comment:
Conex_CCStatus.Set() expects uint8_t but uin16_t is passed. This may cause an overflow. Please fix.